README.md

```diff
@@ -2,15 +2,22 @@
 worker
 ======
 
-The single-cell pipeline work executor.
+The single-cell pipeline work executor. It consists of two containers: a Python container and an R container. 
+
+The Python part of the worker is a wrapper around the R part: it receives tasks from the API, parses them, sends them to the R part for computation, then formats the results, uploads them to S3 and sends a notification to the API that they are ready.
+
+The R part of the worker computes single cell analysis tasks on a pre-processed Seurat rds object. It can communicate only with the Python part of the worker.
 
 ## Running locally
+To run the worker locally, you will need to build it and then run it, passing
+the id of the processed experiment that you want to use the worker with.
 
+### 1. Building the worker
 While in the `worker/` root folder on the host, you can use `make build`.
 
-For example to build and run the r and python containers, you can do:
+To build and run the r and python containers, you can do:
```

Comment:
maybe a little nitpicky, but you used lowercase `r` here while referring it to `R` in the previous paragraph. I think the right name is `r` in uppercase, so it would be good to refer to it in the same way.